### PR TITLE
. : added Procfile.learner

### PR DESCRIPTION
**Pull request by vimalk78**
Added Procfile.learner to easily start cluster with learner

Please read https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: Procfile.learner [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# 2. Add learner node to running cluster
+#   % etcdctl member add infra4 --peer-urls="http://127.0.0.1:42380" --learner=true
+#   Member 4bd62934a5391059 added to cluster ef37ad9dc622a7c4
+
+#   ETCD_NAME="infra4"
+#   ETCD_INITIAL_CLUSTER="infra4=http://127.0.0.1:42380,infra1=http://127.0.0.1:12380,infra2=http://127.0.0.1:22380,infra3=http://127.0.0.1:32380"
+#   ETCD_INITIAL_ADVERTISE_PEER_URLS="http://127.0.0.1:42380"
+#   ETCD_INITIAL_CLUSTER_STATE="existing"
+
+# 3. export the above environment variables
```
Comment (spzala) [inline Procfile.learner]: Nit, for consistency in instructions use `Export` instead of export

File: Procfile.learner [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# 4. start learner node with goreman
```
Comment (spzala) [inline Procfile.learner]: `Start` instead of start

File: Procfile.learner [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+etcd4: bin/etcd --listen-client-urls http://127.0.0.1:42379 --advertise-client-urls http://127.0.0.1:42379 --listen-peer-urls http://127.0.0.1:42380 --initial-cluster-token etcd-cluster-1 --enable-pprof --logger=zap --log-outputs=stderr 
+
+
```
Comment (spzala) [inline Procfile.learner]: Remove extra empty lines?

File: Procfile.learner [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Use goreman to run `go get github.com/mattn/goreman`
+
+# 1. Start the cluster uring Procfile
+# 2. Add learner node to running cluster
```
Comment (spzala) [inline Procfile.learner]: to `a running cluster` ?

**Comment by jingyih** [comment]
To use this script, there are some manual steps needed, as mentioned in the comment section in the script. I feel this script do add some simplification to the process, but also potentially add some user confusion (because by running the script itself is not enough to start a learner node). I think we can add it, if it turns out to be confusing to users we can remove it.

File: Procfile.learner [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Use goreman to run `go get github.com/mattn/goreman`
+
+# 1. Start the cluster uring Procfile
```
Comment (jingyih) [inline Procfile.learner]: using

Comment (vimalk78) [reply Procfile.learner]: updated.

**Comment by vimalk78** [comment]
@spzala @jingyih thanks for your comments. 
updated. PTAL.

**Comment by jingyih** [comment]
lgtm

**Comment by vimalk78** [comment]
> To use this script, there are some manual steps needed, 
I think Procfile can be used to run the command `etcdctl member add infra4 --peer-urls="http://127.0.0.1:42380" --learner=true` also. So there need not be any manual step.
Let me test and get back.

**Comment by jingyih** [comment]
> > To use this script, there are some manual steps needed,
> 
> I think Procfile can be used to run the command `etcdctl member add infra4 --peer-urls="http://127.0.0.1:42380" --learner=true` also. So there need not be any manual step.
> Let me test and get back.

Cool. This sounds good!

**Comment by vimalk78** [comment]
> > Let me test and get back.
> 
> Cool. This sounds good!

`goreman` doesn't start the processes in the order it is defined in the `Procfile`. So even if we add `etcdctl member add` in `Procfile` it is not guaranteed to add the member before starting the learner process. So member addition has to be a manual step. 

I have modified the `Procfile.learner` to include the flags for starting the node, instead of setting env variables. so one manual step is reduced.

File: Procfile.learner (hunk 3 above)
Comment (spzala) [reply Procfile.learner]: @vimalk78 thanks so much for addressing all the comments. We should still remove one extra line here (LOC 9 or 10 :))

Comment (vimalk78) [reply Procfile.learner]: done.

**Comment by spzala** [comment]
LGTM

**Pull request merged**